### PR TITLE
Zen 377 display props

### PR DESCRIPTION
**Pull request by daniel112**
**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-377)

`[semver]: minor`

## Context

* the interface and examples that evf provided in the beginning of this project is updating slowly based on the back and forth we have in the slack channel.
they added a new interface `EFDisplayProperties` which contains:
`displayProperties: {     dateFormat: "MM/dd/yyyy",     timeFormat: EfAgendaTimeFormat.Twelve  }`
* use the `timeFormat` prop to determine whether or not we display our session times in 24hr or 12 hr format

## Goal

* Dynamically set the militaryTime setting based on the `displayProperties.timeFormat` prop

## Updates

* `src/actions/session/mapSessionInterface.js`
    * added `displayProperties` prop to `settings.displayProperties` store
* `src/actions/session/setAgendaSessions.js`
   * set the militaryTime bool based on our store data
* `src/components/sessions.js`
   * ensure we pass in the militaryTime boolean
* `src/mocks/eventsforce/testData.js`
   * added testdata for `displayProperties
## Testing

* Run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-377-display-props`
* navigate to http://local.kegdev.xyz/
* open the `Test Data (JSON)` editor
* change the `displayProperties.timeFormat` to either `"12"` or `"24"`
* hit apply
* verify that the timeblocks and session times update accordingly to use military time or not
* verify the session details modal reflect this change too

### CODE REVIEW

File: src/components/modals/sessionDetailsModal.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ const Body = ({ styles, session }) => {
     ({ items }) => pickKeys(items, ['settings']),
     shallowEqual
   )
-  const { timeFormat } = settings?.agendaSettings?.agendaDisplayProperties
+  const { timeFormat } = settings?.displayProperties || {}
```
Comment (lancetipton) [inline src/components/modals/sessionDetailsModal.js:56]: Just do this =>
```js
const military = settings?.displayProperties?.timeFormat === '24'
```
Then delete `line 57`

File: src/actions/session/setAgendaSessions.js [hunk 2]
```diff
@@ -10,14 +10,16 @@ const { CATEGORIES } = Values
  */
 export const setAgendaSessions = (sessions, agendaDays) => {
   if (!sessions || !agendaDays) return
+  const { items } = getStore()?.getState()
+  const { timeFormat } = items.settings?.displayProperties || {}
```
Comment (lancetipton) [inline src/actions/session/setAgendaSessions.js:14]: Just do this => `const military = items.settings?.displayProperties?.timeFormat === '24'`
Then pass `military` as the last prop to the `buildHourSessionsMap` helper below

**Review by lancetipton — CHANGES_REQUESTED** [review]
Looks good. A couple of small things.

File: src/components/modals/sessionDetailsModal.js (hunk 1 above)
Comment (daniel112) [reply src/components/modals/sessionDetailsModal.js:56]: updated

File: src/actions/session/setAgendaSessions.js (hunk 2 above)
Comment (daniel112) [reply src/actions/session/setAgendaSessions.js]: updated

**Review by lancetipton — APPROVED** [review]
Looks good. Everything worked as expected.

**Review by mpcarolin — APPROVED** [review]
Looks good

**Pull request merged**